Projects/EditCoInsured/Sources/Journeys/EditCoInsuredNavigation.swift

```diff
@@ -0,0 +1,301 @@
+    @Published var showProgressScreenWithoutSuccess = false
+
+    @Published var isEditCoinsuredSelectPresented: InsuredPeopleConfig?
+
+    @Published var externalNavigationRedirect = NavigationPath()
+}
+
+public enum OpenSpecificScreen {
+    case missingAlert
+    case newInsurance
+    case none
+}
+
+public struct EditCoInsuredNavigation: View {
+    let configs: [InsuredPeopleConfig]
+    let onDisappear: () -> Void
+    @State var openSpecificScreen: OpenSpecificScreen
+    @StateObject private var editCoInsuredNavigationVm = EditCoInsuredNavigationViewModel()
+    @StateObject var router = Router()
+
+    public init(
+        configs: [InsuredPeopleConfig],
+        onDisappear: @escaping () -> Void,
+        openSpecificScreen: OpenSpecificScreen? = OpenSpecificScreen.none
+    ) {
+        self.configs = configs
+        self.onDisappear = onDisappear
+        self.openSpecificScreen = openSpecificScreen ?? .none
+
+        let store: EditCoInsuredStore = globalPresentableStoreContainer.get()
+        if let config = editCoInsuredNavigationVm.editCoInsuredConfig {
+            store.coInsuredViewModel.initializeCoInsured(with: config)
+        } else if let config = configs.first {
+            store.coInsuredViewModel.initializeCoInsured(with: config)
+        }
+    }
+
+    public var body: some View {
+        RouterHost(router: router) {
+            if openSpecificScreen == .missingAlert {
+                openMissingCoInsuredAlert()
+            } else if openSpecificScreen == .newInsurance {
+                openNewInsuredPeopleScreen()
+            } else if openSpecificScreen == .none {
+                if configs.count > 1 {
+                    openSelectInsurance(configs: configs)
+                } else if let config = configs.first {
+                    if config.numberOfMissingCoInsuredWithoutTermination > 0 {
+                        if config.fromInfoCard {
+                            openNewInsuredPeopleScreen()
+                        } else {
+                            openRemoveCoInsuredScreen()
+                        }
+                    } else if configs.first?.numberOfMissingCoInsuredWithoutTermination ?? 0 > 0 {
+                        openNewInsuredPeopleScreen()
+                    } else {
+                        openInsuredPeopleScreen()
+                    }
+                }
+            }
+        }
+        .fullScreenCover(item: $editCoInsuredNavigationVm.editCoInsuredConfig) { config in
+            Group {
+                if config.numberOfMissingCoInsuredWithoutTermination > 0 {
+                    openNewInsuredPeopleScreen()
+                } else {
+                    openInsuredPeopleScreen()
+                }
+            }
+            .detent(
+                item: $editCoInsuredNavigationVm.coInsuredInputModel,
+                style: .height
+            ) { coInsuredInputModel in
+                openCoInsuredInput(
```

Comment:
This here and things on line 109 seems same ( only we have .embededInNavigation on the like 125 which we are missing here). Do we need it on 2 places? Also they are looking at the same property --- this will cause problem.

Anyway we can extract this to 
var coInsuredInput: some View {
....
}